doc/api/test.md

```diff
@@ -920,6 +920,61 @@ test('runs timers as setTime passes ticks', (context) => {
 });
 ```
 
+## Snapshot testing
+
+> Stability: 1.0 - Early development
+
+Snapshot tests allow arbitrary values to be serialized into string values and
+compared against a set of known good values. The known good values are known as
+snapshots, and are stored in a snapshot file. Snapshot files are managed by the
+test runner, but are designed to be human readable to aid in debugging. Best
+practice is for snapshot files to be checked into source control along with your
+test files. In order to enable snapshot testing, Node.js must be started with
+the [`--experimental-test-snapshots`][] command-line flag.
+
+Snapshot files are generated by starting Node.js with the
+[`--test-update-snapshots`][] command-line flag. A separate snapshot file is
+generated for each test file. By default, the snapshot file has the same name
+as `process.argv[1]` with a `.snapshot` file extension. This behavior can be
```

Comment:
how does this feature behave in the REPL? should we add a test and/or document that behavior?

Reply:
The docs for `setResolveSnapshotPath()` mention that the path will be `undefined` in that case. There are tests where the path does not resolve to a string (it will throw in that case). If the returned path is a string but not a valid path then it will throw when trying to read/write the file (with the filename attached to the error).